js/ui/overlay/ui.overlay.js

```diff
@@ -878,28 +883,20 @@ const Overlay = Widget.inherit({
         }
     },
 
-    _toggleParentsScrollSubscription: function(subscribe) {
-        if(!this._position) {
-            return;
-        }
-
-        const target = this._position.of || $();
-        const closeOnScroll = this.option('closeOnTargetScroll');
-        let $parents = getElement(target).parents();
+    _toggleParentsScrollSubscription: function(needSubscribe) {
         const scrollEvent = addNamespace('scroll', this.NAME);
+        const parentsScrollSubscriptionInfo = this._parentsScrollSubscriptionInfo;
 
-        if(devices.real().deviceType === 'desktop') {
-            $parents = $parents.add(window);
-        }
-
-        this._proxiedTargetParentsScrollHandler = this._proxiedTargetParentsScrollHandler
-            || (e => { this._targetParentsScrollHandler(e); });
+        eventsEngine.off(parentsScrollSubscriptionInfo.prevTargets, scrollEvent, parentsScrollSubscriptionInfo.handler);
 
-        eventsEngine.off($().add(this._$prevTargetParents), scrollEvent, this._proxiedTargetParentsScrollHandler);
-
-        if(subscribe && closeOnScroll) {
-            eventsEngine.on($parents, scrollEvent, this._proxiedTargetParentsScrollHandler);
-            this._$prevTargetParents = $parents;
+        const closeOnScroll = this.option('closeOnTargetScroll');
+        if(needSubscribe && closeOnScroll) {
+            let $parents = getElement(this._position.of).parents();
```

Comment:
why do we get parents from position.of. not container?

Reply:
reworked. at present, we get parents of wrapper element